cep-????.md

```diff
@@ -0,0 +1,129 @@
+# CEP ???? - Virtual packages
+
+<table>
+<tr><td> Title </td><td> Virtual packages </td>
+<tr><td> Status </td><td> Draft </td></tr>
+<tr><td> Author(s) </td><td> Jaime Rodríguez-Guerra &lt;jaime.rogue@gmail.com&gt;</td></tr>
+<tr><td> Created </td><td> Dec 17, 2024</td></tr>
+<tr><td> Updated </td><td> Dec 17, 2024</td></tr>
+<tr><td> Discussion </td><td> https://github.com/conda/ceps/pull/103 </td></tr>
+<tr><td> Implementation </td><td> https://github.com/conda/conda/tree/24.11.1/conda/plugins/virtual_packages, https://github.com/mamba-org/mamba/blob/libmamba-2.0.5/libmamba/src/core/virtual_packages.cpp, https://github.com/conda/rattler/tree/rattler-v0.28.8/crates/rattler_virtual_packages/src </td></tr>
+</table>
+
+> The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT",
+  "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as
+  described in [RFC2119][RFC2119] when, and only when, they appear in all capitals, as shown here.
+
+## Abstract
+
+This CEP standardizes which virtual packages MUST be offered by conda solvers.
+
+## Specification
+
+A virtual package is defined as a package record with three fields: name, version and build string.
+The name MUST start with double underscore. The version and build string MUST follow the same semantics as in regular package records.
+
+### List of virtual packages
+
+In alphabetical order, every conda client MUST support the following virtual packages:
+
+- `__archspec`
+- `__cuda`
+- `__glibc`
+- `__linux`
+- `__osx`
+- `__unix`
+- `__win`
+
+#### `__archspec`
+
+This virtual package MUST be always present, with the version set to `1`. The build string SHOULD reflect the detected CPU microarchitecture. If it cannot be detected, the build string SHOULD be `0`.
+
+The build string MUST be overridable with the `CONDA_OVERRIDE_ARCHSPEC` environment variable, if set to a non-empty value.
+
+#### `__cuda`
+
+This virtual package MUST be present when the system exhibits GPU drivers compatible with the CUDA runtimes. When available, the version value MUST be set to the oldest CUDA version supported by the detected drivers (i.e. the formatted value of `libcuda.cuDriverGetVersion()`), constrained to the first two components (major and minor) and formatted as `{major}.{minor}`. The build string MUST be `0`.
+
+The version MUST be overridable with the `CONDA_OVERRIDE_CUDA` environment variable, if set to a non-empty value.
+
+#### `__glibc`
+
+This virtual package MUST be present when the native platform is `linux-*`. Its version value MUST be set to the system `libc` version, constrained to the first two components (major and minor) formatted as `{major}.{minor}`. The build string MUST be `0`.
```

Comment:
In case we ever add `__musl`, etc.

```suggestion
This virtual package MUST be present when the native platform is `linux-*`. Its version value MUST be set to the system GNU `libc` version, constrained to the first two components (major and minor) formatted as `{major}.{minor}`. The build string MUST be `0`.
```